libs/langchain_v1/langchain/agents/middleware_agent.py

```diff
@@ -331,9 +355,11 @@ def model_request(state: dict[str, Any]) -> dict[str, Any]:
 
         # Apply modify_model_request middleware in sequence
         for m in middleware_w_modify_model_request:
-            # Filter state to only include fields defined in this middleware's schema
-            filtered_state = _filter_state_for_schema(state, m.state_schema)
-            request = m.modify_model_request(request, filtered_state)
+            sig = signature(m.modify_model_request)
```

Comment:
does `signature` do anything expensive? And is this temporary or is it to support middleware that doesn't match the interface?